### ISSUE TITLE
Keep generated contents in sync with Base localization values

### ISSUE THREAD

**Issue by zjx20** [Flinesoft/BartyCrouch#4]
For example, there is one generated comment:
```
/* Class = "UITableViewSection"; headerTitle = "Foo"; ObjectID = "HQs-Z9-d7P"; */
"HQs-Z9-d7P.headerTitle" = "Foo";
```

If the base localization "Foo" was changed to "Bar" in the base `Main.storyboard`, can `.strings` files be also updated to reflect the change? To become:
```
/* Class = "UITableViewSection"; headerTitle = "Bar"; ObjectID = "HQs-Z9-d7P"; */
"HQs-Z9-d7P.headerTitle" = "Bar";  /* gets the update too if it wasn't a translated value */
```

**Comment by Dschee** [comment]
In your example you both update the comment and actual value. So, just so I get this right, your idea is to keep both the comment and value updated so long as it was never changed from the generated defaults to something else?

So this:
``` Strings
/* Class = "UITableViewSection"; headerTitle = "Foo1"; ObjectID = "HQs-Z9-d7P"; */
"HQs-Z9-d7P.headerTitle" = "Foo1";
/* Class = "UITableViewSection"; headerTitle = "Foo2"; ObjectID = "Hk7-27-Eff"; */
"Hk7-27-Eff.headerTitle" = "TranslatedFoo2";
/* Class = "UITableViewSection"; headerTitle = "ChangedFoo3"; ObjectID = "nKk-K2-Zsh"; */
"nKk-K2-Zsh.headerTitle" = "Foo3";
```
Would become that if all Base values were multiplied by 100:
``` Strings
/* Class = "UITableViewSection"; headerTitle = "Foo100"; ObjectID = "HQs-Z9-d7P"; */
"HQs-Z9-d7P.headerTitle" = "Foo100";
/* Class = "UITableViewSection"; headerTitle = "Foo200"; ObjectID = "Hk7-27-Eff"; */
"Hk7-27-Eff.headerTitle" = "TranslatedFoo2";
/* Class = "UITableViewSection"; headerTitle = "ChangedFoo3"; ObjectID = "nKk-K2-Zsh"; */
"nKk-K2-Zsh.headerTitle" = "Foo300";
```

Is this what you would expect? That does seem to make sense as the default behavior if the option `--default-to-base` is set. I also think it makes sense for the comment part alone even if it doesn't default to base.

Maybe we should make this the default behavior and introduce two separate options `--disable-comment-update` and `--disable-base-update` which turn off the respective update features once somebody has a use case for this to be deactivated. So like the `addNewValuesAsEmpty` before the code for disabling this should already be there but no CLI for it present until it's needed.

Would this be what you asked for?

**Comment by zjx20** [comment]
I didn't quite understand the third case in your example, I thought it should be "Foo300" in the comment part too. In my opinion, comments should always keep in sync with the Base, so that you definitely know what you are translating.

In fact, it is impossible to keep it as "ChangedFoo3". To achieve that, we have to know whether it has been changed. But we just know the current Base value is "Foo300", and we have no idea what the original value was. So no matter what values in the comment part ("Foo3" or "ChangedFoo3"), we always treat it as a changed value.

> That does seem to make sense as the default behavior if the option --default-to-base is set. I also think it makes sense for the comment part alone even if it doesn't default to base.

> Maybe we should make this the default behavior and introduce two separate options --disable-comment-update and --disable-base-update which turn off the respective update features once somebody has a use case for this to be deactivated. So like the addNewValuesAsEmpty in #2 the code for disabling this should already be there but no CLI for it present until it's needed.

I agree with you.

**Comment by zjx20** [comment]
Should we assume that no one will change the comment part? So if the actual value is same as the commented one, we know the actual value is untranslated.

**Comment by Dschee** [comment]
Yeah, you've got a point about the third example, we can't know the previous base value without saving it somewhere in a temporary file (which I think we shouldn't do just for this small feature as it has quite some overhead).

About assuming that the comment part will not be changed:
I'd like to keep BartyCrouch as flexible as possible so it can be widely adopted which is why I don't think we should assume too much here. But having thought this twice I actually feel that we could state in the docs that comments will be auto-updated so long as the structure of the comment isn't changed. Currently this would mean so long as the comment meats the following structure (it is what ibtool generates) they will be auto-updated: `/* Class = "<...>"; <...> = "<BASE_VALUE>"; ObjectID = "<...>"; */`.

I think the number of people who keep this structure and only update the `<BASE_VALUE>` part updated manually is near zero and that's not a good solution anyway. So if those are seeking for a better solution they should learn from BartyCrouch that there's a better way (if they still want to keep updating them manually). This is why I think we need to also recommend a way of changing that structure for those who want to update their values manually (I'm against introducing a command line option to turn this off, I'm in favor of keeping the options clean, at least until there seems to be good reason for it).

The simplest recommendation I can come up with is to actually just add a `!` to the opening `/*` comment resulting to an opening `/*!`. It's like `/*` tells Xcode to start ignoring and `/*!` yells at Xcode to start ignoring so Xcode can cause BartyCrouch to do so, too. ^^

What do you think?

**Comment by zjx20** [comment]
I think your `/*!` solution is good enough, I'm happy with that!

**Comment by Dschee** [comment]
I just released v1.3.0 which should cover all features requested/discussed here. Hope it helps!

**Comment by zjx20** [comment]
Cool! Thank you very much!
